llvm/test/CodeGen/AMDGPU/insert-singleuse-vdst.mir

```diff
@@ -398,7 +398,6 @@ body: |
   ; CHECK-NEXT:   successors: %bb.1(0x80000000)
   ; CHECK-NEXT:   liveins: $sgpr0_sgpr1
   ; CHECK-NEXT: {{  $}}
-  ; CHECK-NEXT:   S_SINGLEUSE_VDST 1
```

Comment:
Why did this change? I don't see any wide register or subregister accesses related to the def of vgpr0 in this test.

Reply:
I should have mentioned this in the commit message. This test should not be marked as single use as the use is after an exec mask change. This is caused by the change in AMDGPUInsertSingleUseVDST.cpp on line 110 replacing the RegisterUseCount.erase(Reg) with RegisterUseCount[Unit] = 0. When an exec mask change is detected, all registers in RegisterUseCount are given a value of 2 uses to exclude them from being marked as single use producers (line 92). However, when the values were being erased, they were no longer being correctly excluded. Setting the use count to 0 and keeping them in RegisterUseCount prevents this.

Reply:
Could you split that fix out into a separate PR please?

Reply:
Sure, no problem.